azext_iot/iothub/common.py

```diff
@@ -0,0 +1,68 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+shared: Define shared data types(enums) and constant strings.
+"""
+
+from enum import Enum
+
+
+SYSTEM_ASSIGNED_IDENTITY = '[system]'
+
+
+class AuthenticationType(Enum):
+    """
+    Type of the Authentication for the routing endpoint.
+    """
+    KeyBased = 'keyBased'
+    IdentityBased = 'identityBased'
+
+
+class EncodingFormat(Enum):
+    """
+    Type of the encoding format for the container.
+    """
+    JSON = 'json'
+    AVRO = 'avro'
+
+
+class EndpointType(Enum):
+    """
+    Type of the routing endpoint.
+    """
+    EventHub = 'eventhub'
+    ServiceBusQueue = 'servicebus-queue'
+    ServiceBusTopic = 'servicebus-topic'
+    AzureStorageContainer = 'storage-container'
+    CosmosDBCollection = 'cosmosdb-collection'
+
+
+class IdentityType(Enum):
+    """
+    Type of managed identity for the IoT Hub.
+    """
+    system_assigned = "SystemAssigned"
+    user_assigned = "UserAssigned"
+    system_assigned_user_assigned = "SystemAssigned, UserAssigned"
+    none = "None"
+
+
+class RouteSourceType(Enum):
+    """
+    Type of the route source.
+    """
+    # Invalid = 'invalid'
```

Comment:
this was in the azure cli core endpoint types but this source type just errors out. Can we get rid of it?

Reply:
I wonder if this value was just used to map from all possible return types from the service/SDK - I don't see why we'd need this value from the client perspective, no reason to create/test all routes with `invalid` types.

https://github.com/Azure/azure-cli/blob/99f07bc11069b66ed7e4dfbd01b86868371ee9ac/src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py#L365

Reply:
ok will leave it in just incase some dev wants it but will add a class function to return a list without the invalid:
```
    @classmethod
    def list_valid_types(cls):
        return list(filter(lambda d: d != RouteSourceType.Invalid.value, map(lambda c: c.value, cls)))```